Author: Bernie Innocenti <bernie@codewiz.org>

Fix the build harness for contemporary versions of the autoshite suite.

We can't invoke libtool from within configure because it's not yet there;
but we don't need libtool if we use pkg-config to detect libraries.

This change opens the possibility to drop a bunch of macros from
unix/sidplay2.m4 and simplify other things, but this patch doesn't
bother to.


--- sidplay-2.0.9.old/configure.ac	2009-09-06 16:11:38.362313298 -0400
+++ sidplay-2.0.9/configure.ac	2009-09-06 16:15:34.224189943 -0400
@@ -1,6 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(Makefile.am)
 AC_CONFIG_AUX_DIR(unix)
+AC_CONFIG_MACRO_DIR(unix)
 AM_CONFIG_HEADER(unix/config.h)
 
 dnl Setup library CURRENT, REVISION and AGE
@@ -114,16 +115,15 @@ MY_CHECK_IOS_BIN
 MY_CHECK_IOS_OPENMODE
 
 dnl Check for libraries
-SID2_LIB_CHECKS
-SID2_FIND_LIBSIDPLAY2
-SID2_FIND_LIBSIDUTILS
+PKG_CHECK_MODULES(LIBSIDPLAY2, libsidplay2)
+PKG_CHECK_MODULES(LIBSIDUTILS, libsidutils)
 
 # Find sid builder emulations
 SID2_FIND_BUILDERS
 
 dnl Add absolute directory for header files
 CXXFLAGS="$CXXFLAGS -I\$(top_srcdir)/include \
-\$(LIBSIDPLAY2_CXXFLAGS) \$(LIBSIDUTILS_CXXFLAGS)"
+\$(LIBSIDPLAY2_CFLAGS) \$(LIBSIDUTILS_CFLAGS)"
 
 ACLOCAL_AMFLAGS="-I unix"
 AC_SUBST(ACLOCAL_AMFLAGS)
diff -rup sidplay-2.0.9.old/src/Makefile.am sidplay-2.0.9/src/Makefile.am
--- sidplay-2.0.9.old/src/Makefile.am	2003-07-07 15:58:53.000000000 -0400
+++ sidplay-2.0.9/src/Makefile.am	2009-09-06 16:10:21.236220894 -0400
@@ -11,7 +11,7 @@ player.cpp player.h
 
 sidplay2_LDADD = ./audio/null/libnull.a ./audio/wav/libwav.a \
 $(AUDIO_DRV) $(AUDIO_LDFLAGS) \
-$(LIBSIDPLAY2_LDFLAGS) $(LIBSIDUTILS_LDFLAGS) \
+$(LIBSIDPLAY2_LIBS) $(LIBSIDUTILS_LIBS) \
 $(BUILDERS_LDFLAGS)
 
 # Remove bad default includes
--- sidplay-2.0.9/unix/my_macros.m4	2009-09-07 08:48:22.441381197 -0400
+++ sidplay-2.0.9.bernie/unix/my_macros.m4	2009-09-06 11:51:18.198661349 -0400
@@ -163,7 +163,7 @@ AC_DEFUN([MY_TRY_COMPILE],
 
     CXXFLAGS="$CXXFLAGS $1"
     LDFLAGS="$LDFLAGS $2"
-    CXX="${SHELL-/bin/sh} ${srcdir}/libtool $CXX"
+    #CXX="${SHELL-/bin/sh} ${srcdir}/libtool $CXX"
 
     AC_TRY_LINK(
         [#include <$3>],
